php-checks/src/main/java/org/sonar/php/checks/phpunit/TestCatchRuntimeExceptionCheck.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks.phpunit;
+
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.CheckUtils;
+import org.sonar.php.checks.utils.PhpUnitCheck;
+import org.sonar.php.symbols.Symbols;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.declaration.NamespaceNameTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+import org.sonar.plugins.php.api.tree.expression.LiteralTree;
+import org.sonar.plugins.php.api.tree.expression.MemberAccessTree;
+import org.sonar.plugins.php.api.tree.statement.CatchBlockTree;
+
+import static org.sonar.php.checks.utils.CheckUtils.trimQuotes;
+import static org.sonar.plugins.php.api.symbols.QualifiedName.qualifiedName;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.CLASS_MEMBER_ACCESS;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.NAMESPACE_NAME;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.REGULAR_STRING_LITERAL;
+
+@Rule(key = "S3477")
+public class TestCatchRuntimeExceptionCheck extends PhpUnitCheck {
+
+  private static final String MESSAGE = "Remove this catch block so that error conditions can be caught in testing, rather than in the field.";
+  private static final String EXPECT_EXCEPTION_ASSERTION = "expectException";
+  private static final String RUNTIME_EXCEPTION = "RuntimeException";
+
+  @Override
+  public void visitCatchBlock(CatchBlockTree tree) {
+    if (isPhpUnitTestMethod()) {
+      tree.exceptionTypes().stream()
+        .filter(this::isRuntimeException)
+        .forEach(this::addIssue);
+    }
+
+    super.visitCatchBlock(tree);
+  }
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
```

Comment:
After re-reading the rule description I think that it is actually only relevant for catches in try-catch. If a developer would use "expectException" he explicitly awaits such an exception and will notice if it is not thrown  (the test will fail). However, in a try-catch it is not explicit and it might not be thrown. So if there is an `expectException(RuntimeException::class)` it is actually a good thing. I think doing something like the following would be good:
- Only `expectException(RuntimeException::class)`  -> good.
- Only `catch(RuntimeException $e)` -> bad. 
- `expectException(RuntimeException::class)` and `catch(RuntimeException $e)` -> good.
- `catch(RuntimeException $e)` and a call to `$this->fail()` in the try -> good . In that way it is usually done to notice when the exception is not thrown (see first case in the ITS results you have added.).